cmd/juju/application/deployer/charm.go

```diff
@@ -584,6 +592,40 @@ func (c *repositoryCharm) PrepareAndDeploy(ctx *cmd.Context, deployAPI DeployerA
 	return c.deploy(ctx, deployAPI)
 }
 
+func (c *repositoryCharm) uploadExistingPendingResources(
```

Comment:
having this method here, means there are similar code paths in this file and resource.go which is confusing. can it be cmd/juju/application/deployer/resource.go instead? then sharing with bundles will be easier in the future.